Sources/GraphQLLanguage/Builder.swift

```diff
@@ -10,24 +10,36 @@ import Foundation
 
 enum BuildError: Error, CustomDebugStringConvertible {
     case unwrap
-    case unexpectedContext(_ context: ParserRuleContext)
+    case unexpectedContext(text: String, token: Token?)
 
     // TODO: Improve error description
     var debugDescription: String {
         switch self {
         case .unwrap:
             return "Required field must not be nil. Probably a parser bug."
-        case .unexpectedContext(let context):
-            let rule = context.toStringTree(GraphQLParser.ruleNames)
-            if let token = context.getStart() {
-                let line = token.getLine()
-                let position = token.getCharPositionInLine()
-                return "Build error at: \(rule) line:\(line) position:\(position)"
-            } else {
-                return "Build error at: \(rule)"
-            }
+        case .unexpectedContext(let text, let token):
+            return "Build error: \(text)" + (token.map { token in
+                " at line:\(token.getLine()) position:\(token.getCharPositionInLine())"
+            } ?? "")
         }
     }
+
+    /// A `ParserRuleContext` is only valid so long as its associated `Document`
+    /// exists. Because of this caveat, the components relevant to the
+    /// `.unexpectedContext` error must be captured before the context is invalid.
+    ///
+    /// This convenience initializer will capture the relevant context properties and
+    /// populate an `.unexpectedContext` error with them.
+    ///
+    /// - Warning: If this method is called after the `context`'s `Document` is
+    /// released, the program will crash.
+    ///
+    /// - Returns: A `BuildError.unexpectedContext` constructed from
+    /// the given `ParserRuleContext`.
+    static func unexpectedContext(from context: ParserRuleContext) -> BuildError {
```

Comment:
Probably this is the rare case though, likely this is a valid case to use `fileprivate` instead of module visibility?

Reply:
Yeah, that makes sense!